nsqd/nsqd.go

```diff
@@ -203,13 +206,14 @@ func (n *NSQD) getFlag(f int32) bool {
 }
 
 func (n *NSQD) SetHealth(err error) {
-	n.errMtx.Lock()
-	defer n.errMtx.Unlock()
-	n.err = err
 	if err != nil {
 		n.setFlag(flagHealthy, false)
+		n.errValue.Store(errStore{err: err})
```

Comment:
Unfortunately you can't call `atomic.Value' 'Store()` with `nil` ([doc](https://golang.org/pkg/sync/atomic/#Value)). Even if we stored just the last value, and relied on flagHealthy, they need to be of the same concrete type.